src/editors/mongoDBDocumentService.ts

```diff
@@ -0,0 +1,154 @@
+    statusView: StatusView,
+    telemetryController: TelemetryController
+  ) {
+    this._context = context;
+    this._documentIdStore = documentIdStore;
+    this._connectionController = connectionController;
+    this._statusView = statusView;
+    this._telemetryController = telemetryController;
+  }
+
+  _fetchDocumentFailed(message: string): void {
+    const errorMessage = `Unable to fetch document: ${message}`;
+
+    throw new Error(errorMessage);
+  }
+
+  _saveDocumentFailed(message: string): void {
+    const errorMessage = `Unable to save document: ${message}`;
+
+    // Send a telemetry event that saving the document failed.
+    this._telemetryController.trackDocumentUpdated('treeview', false);
+
+    throw new Error(errorMessage);
+  }
+
+  async replaceDocument(data: {
+    documentId: EJSON.SerializableTypes;
+    namespace: string;
+    connectionId: string;
+    newDocument: EJSON.SerializableTypes;
+  }): Promise<void> {
+    log.info('replace document in MongoDB', data);
+
+    const { documentId, namespace, connectionId, newDocument } = data;
+    const activeConnectionId = this._connectionController.getActiveConnectionId();
+    const connectionName = this._connectionController.getSavedConnectionName(
+      connectionId
+    );
+
+    if (activeConnectionId !== connectionId) {
+      return this._saveDocumentFailed(
+        `no longer connected to '${connectionName}'`
+      );
+    }
+
+    const dataservice = this._connectionController.getActiveDataService();
+
+    if (dataservice === null) {
+      return this._saveDocumentFailed(
+        `no longer connected to '${connectionName}'`
+      );
+    }
+
+    const findOneAndReplace = util.promisify(
+      dataservice.findOneAndReplace.bind(dataservice)
+    );
+
+    this._statusView.showMessage('Saving document...');
+
+    try {
+      await findOneAndReplace(
+        namespace,
+        {
+          _id: documentId
+        },
+        newDocument,
+        {
+          returnOriginal: false
+        }
+      );
+
+      this._statusView.hideMessage();
+      this._telemetryController.trackDocumentUpdated('treeview', true);
+    } catch (error) {
+      this._statusView.hideMessage();
+
+      return this._saveDocumentFailed(error.message);
+    }
+  }
+
+  async fetchDocument(data: {
+    documentId: EJSON.SerializableTypes;
+    namespace: string;
+  }): Promise<EJSON.SerializableTypes | void> {
+    log.info('fetch document from MongoDB', data);
+
+    const { documentId, namespace } = data;
+    const activeConnectionId = this._connectionController.getActiveConnectionId();
+    const connectionName = activeConnectionId
+      ? this._connectionController.getSavedConnectionName(activeConnectionId)
+      : '';
+    const dataservice = this._connectionController.getActiveDataService();
+
+    if (dataservice === null) {
+      return this._fetchDocumentFailed(
+        `no longer connected to ${connectionName}`
```

Comment:
I think connection name we show here will always be an empty string on a disconnected connection - is it possible to show the connection name of the connection which the playground result was run against?

Reply:
I will address this in the next PR, thx for pointing at it!